File: src/main/java/com/aliiensmp/aliienResize/Economy/currencyOptions/Vault.java
package com.aliiensmp.aliienResize.Economy.currencyOptions;

import com.aliiensmp.aliienResize.AliienResize;
import com.aliiensmp.aliienResize.Economy.CurrencyProvider;
import org.bukkit.entity.Player;

public class Vault implements CurrencyProvider {

    private final AliienResize plugin;
    private final String suffix;

    /**
     * Creates the Vault currency adapter.
     *
     * @param plugin owning plugin instance
     */
    public Vault(AliienResize plugin, String suffix) {
        this.plugin = plugin;
        this.suffix = suffix;
    }

    @Override
    public boolean isValid() {
        return plugin.getVaultExpansion() != null && plugin.getVaultExpansion().hasEconomy();
    }

    @Override
    public boolean hasBalance(Player player, double amount) {
        if (!isValid()) return false;
        return plugin.getVaultExpansion().getEconomy().has(player, amount);
    }

    @Override
    public boolean withdraw(Player player, double amount) {
        if (!hasBalance(player, amount)) return false;
        return plugin.getVaultExpansion().getEconomy().withdrawPlayer(player, amount).transactionSuccess();
    }

    @Override
    public String suffix() {
        return suffix;
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/economy/currencyOptions/Vault.kt
package com.aliiensmp.aliienResize.economy.currencyOptions

import AliienResize
import com.aliiensmp.aliienResize.economy.CurrencyProvider
import org.bukkit.entity.Player

class Vault(private val plugin: AliienResize, override val suffix: String) : CurrencyProvider {

    override val isValid: Boolean
        get() = plugin.vaultExpansion?.hasEconomy == true

    override fun hasBalance(player: Player, amount: Double): Boolean {
        if (!isValid) return false

        return plugin.vaultExpansion?.economy?.has(player, amount) == true
    }

    override fun withdraw(player: Player, amount: Double): Boolean {
        if (!hasBalance(player, amount)) return false

        return plugin.vaultExpansion?.economy?.withdrawPlayer(player, amount)?.transactionSuccess() == true
    }
}